src/main/scala/com/scylladb/migrator/readers/Cassandra.scala

```diff
@@ -126,7 +128,17 @@ object Cassandra {
 
       timestampsToFields
         .map {
-          case ((ttl, writetime), fields) =>
+          case ((ttl, writetime), fields) => {
+            val writetimestamp = writetime.getOrElse(CassandraOption.Unset)
+            val baseTTL = if (defaultTTL > 0) defaultTTL else 0L
+            val deltaTTL = if (writetimestamp == CassandraOption.Unset) {
+              0
+            } else {
+              baseTTL - (System.currentTimeMillis - writetimestamp.asInstanceOf[Long] / 1000) / 1000
+            }
```

Comment:
I recommend to avoid using `asInstanceOf`. I suggest the following changes, which I believe are more readable:

```suggestion
            val deltaTTL = writetime match {
              case None => 0
              case Some(writetimestamp) =>
                baseTTL - (System.currentTimeMillis - writetimestamp / 1000) / 1000
            }
```